pkg/liquidity-mining/test/AuthorizerAdaptorEntrypoint.test.ts

```diff
@@ -0,0 +1,166 @@
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import { expect } from 'chai';
+import { defaultAbiCoder } from 'ethers/lib/utils';
+import { sharedBeforeEach } from '@balancer-labs/v2-common/sharedBeforeEach';
+import { ANY_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+
+describe('AuthorizerAdaptorEntrypoint', () => {
+  let vault: Contract;
+  let authorizer: Contract;
+  let adaptor: Contract;
+  let entrypoint: Contract;
+  let paymentReceiver: Contract;
+  let admin: SignerWithAddress, grantee: SignerWithAddress, other: SignerWithAddress;
+
+  before('setup signers', async () => {
+    [, admin, grantee, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy vault with entrypoint', async () => {
+    ({ instance: vault, authorizer, authorizerAdaptor: adaptor } = await Vault.create({ admin }));
+
+    // TODO(@jubeira): initialize entrypoint and adaptor inside helpers.
+    entrypoint = await deploy('AuthorizerAdaptorEntrypoint', { args: [adaptor.address] });
+    await authorizer.setAdaptorEntrypoint(entrypoint.address);
+  });
+
+  sharedBeforeEach('deploy mock to receive payments', async () => {
+    paymentReceiver = await deploy('MockPaymentReceiver');
+  });
+
+  describe('constructor', () => {
+    it('sets the vault address', async () => {
+      expect(await entrypoint.getVault()).to.be.eq(vault.address);
+    });
+
+    it('uses the authorizer of the vault', async () => {
+      expect(await entrypoint.getAuthorizer()).to.equal(authorizer.address);
+    });
+
+    it('uses the adaptor of the vault', async () => {
+      expect(await entrypoint.getAuthorizerAdaptor()).to.equal(adaptor.address);
+    });
+
+    it('returns the same action ID as the adaptor', async () => {
+      expect(await entrypoint.getActionId('0xaabbccdd')).to.equal(await adaptor.getActionId('0xaabbccdd'));
+    });
+
+    it('tracks authorizer changes in the vault', async () => {
+      const action = await actionId(vault, 'setAuthorizer');
+      await authorizer.connect(admin).grantPermissions([action], admin.address, [ANY_ADDRESS]);
+
+      await vault.connect(admin).setAuthorizer(other.address);
```

Comment:
Yes, that's already done in the subsequent PR #1981.

We did it this way to split the work in somewhat smaller chunks; once you move that to the helper you get a lot of failures since the adaptor stops working by itself without the entrypoint. All the TODOs were a workaround to be able to do this split, and shall be removed in #1981.

See https://github.com/balancer-labs/balancer-v2-monorepo/pull/1977#issuecomment-1302263630.

Reply:
The change in #1981 isn't quite what I was meaning here. That handles the initial deployment of the vault and authorizer, etc. whereas I mean a generic wrapper around the `setAuthorizer` function that we can call from tests like this.

i.e. we could add this function to the Vault model
```
  async setAuthorizer(newAuthorizer: Contract): Promise<ContractTransaction> {
    const action = await actionId(this.instance, 'setAuthorizer');
    await this.grantPermissionsGlobally([action], this.admin);
    return vault.connect(admin).setAuthorizer(newAuthorizer.address);
  }
```